apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -2,21 +2,48 @@ defmodule GameBackend.Matches do
   @moduledoc """
   Matches
   """
+  alias GameBackend.Users
+  alias GameBackend.Users.Currencies
   alias Ecto.Multi
   alias GameBackend.Repo
   alias GameBackend.Matches.ArenaMatchResult
 
-  def create_arena_match_results(match_id, results) do
-    transaction = Multi.new()
+  def create_arena_match_results(results) do
+    {:ok, conrrency_config_json} =
+      Application.app_dir(:game_backend, "priv/currencies_rules.json")
+      |> File.read()
 
-    transaction =
-      Enum.reduce(results, transaction, fn result, transaction_acc ->
-        attrs = Map.put(result, "match_id", match_id)
-        changeset = ArenaMatchResult.changeset(%ArenaMatchResult{}, attrs)
-        transaction_acc = Multi.insert(transaction_acc, {:insert, attrs["user_id"]}, changeset)
-        transaction_acc
-      end)
+    currency_config =
+      Jason.decode!(conrrency_config_json)
 
-    Repo.transaction(transaction)
+    Enum.reduce(results, Multi.new(), fn result, transaction_acc ->
+      changeset = ArenaMatchResult.changeset(%ArenaMatchResult{}, result)
+      {:ok, google_user} = Users.get_google_user(result["user_id"])
+
+      amount_of_trophies = Currencies.get_amount_of_currency_by_name(google_user.user.id, "Trophies")
```

Comment:
Not to fix in this PR, but is it expected/right to need that `google_user.user`? Shouldn't the currency be in the meta-user?

Reply:
Imo the currencies should be in the User, since it's different for each game, in the future we can have a feature to move currencies between users bellow the same meta-users so we can have gold in curse and move it to champions a rate